rewards-distribution/ethereum/client/README.md

```diff
@@ -1,11 +1,27 @@
 # Rewards Distribution using OrbsRewardsDistribution contract
 
-* Prerequisute - address of a deployed `OrbsRewardsDistribution` contract
-* In a terminal, navigate to the root of truffle project under `rewards-distribution/ethereum`
-* execute the batch generation script:
+Used as an infrastructure to distribute rewards, it will
+* Take as an input a csv file holding address and amount (in full orbs) to distribute to
+* Create batches of distribution to update the contract with
+* Distribute after batches has been set up
+
+In order to use it, you will first need to
+* Have the address of a deployed `OrbsRewardsDistribution` contract
+* Have a csv file with the distribution details
+* Have the distribution event name
+* Run npm install, have truffle config up to date and relevant to your system
+* Ensure that the `truffle-config` is configured and that all environment variables are valid (ethereum url, mnemonic and so on)
+* **Check the gas prices and adjust it in the truffle config**
+
+## Creating the batches
+To begin working, in a terminal, navigate to the root of truffle project under `rewards-distribution/ethereum`
+You will need to then execute the batch generation script:
 ```$bash
-truffle exec client/getBatchHahses.js  [csv filename] [number of payments per batch]
+./node_modules/.bin/truffle exec client/getBatchHahses.js  [csv filename] [number of payments per batch] --network ropsten/mainnet/development
```

Comment:
consider adding npm commands so that the instructions will read:
`npm truffle exec...`

Reply:
consider replacing `ropsten/mainnet/development` with `[ropsten|mainnet|development]`